src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -296,9 +307,9 @@ public void execute() {
             log("compiled on " + compileTimestamp, Project.MSG_VERBOSE);
 
             // Check for no arguments
-            if (fileName == null && fileSets.isEmpty()) {
+            if (fileName == null && fileSets.isEmpty() && sourcePaths.isEmpty()) {
```

Comment:
@MEZk , @rnveach 
It is very uncertain point of refactoring, sometime it is required and sometimes it clear in single line.
we could start collecting use cases where refactoring is beneficial.

This is definitely Check for sevntu ..... as there will be bunch of nuances, that we do not know  for now. It is unlikely that first implementation become good to use. All such brave ideas we always tested on own code and then in proprietary code to let other agree with Check. As some long time passed and no negative feedback from non-chekstyle users - we could think to move it to main project.

One visible nuance is that multi-line expressions will take more vertical space so vertical size will increase, it might be a conflict with anonymous class length Check or ...... .